types/sinon/index.d.ts

```diff
@@ -774,7 +774,9 @@ declare namespace Sinon {
         (obj: any): SinonMock;
     }
 
-    type SinonFakeTimers = FakeTimers.Clock & {
+    type SinonFakeNodeClock = FakeTimers.NodeClock;
+    type SinonFakeBrowserClock = FakeTimers.BrowserClock;
+    type SinonFakeTimers<Clock extends FakeTimers.FakeClock<any> = FakeTimers.Clock> = Clock & {
```

Comment:
what are you trying to achieve here?

Reply:
Ergonomic typings. Otherwise we'd have to import types from `@sinonjs/fake-timers` to get the correct clock. [`74c8c93` (#57290)](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/57290/commits/74c8c93be20312a38f857bf14f3414b7f37f6aa9) shows the effect this had on the usage.

Reply:
the `any` is whats suspect here, its really a hack to allow all types of a generic because you don't want to have to specify what its type parameter is.

i think it'd be nicer if you can avoid that:

```ts
type SinonFakeTimers<T extends FakeTimers.TimerId = FakeTimers.TimerId> = FakeTimers.Clock<T> & {
```

Reply:
> its really a hack to allow all types of a generic because you don't want to have to specify what its type parameter is.

I'm usually not concerned with what's a "hack" or not but rather what works. Could you elaborate what the specific problem here is other than "I don't like `any`"?

> because you don't want to have to specify what its type parameter is.

I do want though. 

> i think it'd be nicer if you can avoid that:

Which would require passing the TimerId type instead which I find less ergonomic than passing the full clock type (`NodeClock` vs `BrowserClock`). It's also more descriptive.

Reply:
Actually I'm not interested in this change.

Reply:
think of it like having an array of generics:

```ts
const arr: Array<Clock<T>> = [
  Clock<T1>,
  Clock<T2>
];
```

typing this is `Clock<any>` will make it work but is effectively turning type checking off. to have a "proper" type, it would have to be `Array<Clock<T1> | Clock<T2>>`.  or you'd implement some `ClockLike` non-generic interface all generic clocks implement.

the `any` is a workaround having to specify the individual types.